websockets/kiturawebsockets.html

```diff
@@ -0,0 +1,315 @@
+        </ul>
+        <li class="sidebar-item collapsible">Web Application</li>
+        <ul class="nested-sidebar-list content">
+          <li class="nested-sidebar-item"><a href="../templating/templating.html">What are Web Applications?</a></li>
+          <li class="nested-sidebar-item"><a href="../templating/static-file-server.html">Static File Server</a></li>
+          <li class="nested-sidebar-item"><a href="../templating/stencil.html">Stencil</a></li>
+          <li class="nested-sidebar-item"><a href="../templating/markdown.html">Markdown</a></li>
+        </ul>
+        <li class="sidebar-item collapsible">WebSockets</li>
+        <ul class="nested-sidebar-list content">
+          <li class="nested-sidebar-item"><a href="../websockets/websockets.html">What are WebSockets?</a></li>
+          <li class="nested-sidebar-item"><a href="../websockets/kiturawebsockets.html">Kitura-WebSocket Echo Server</a></li>
+        </ul>
+        <li class="sidebar-item collapsible">Deploying</li>
+        <ul class="nested-sidebar-list content">
+          <li class="nested-sidebar-item"><a href="../deploying/monitoring.html">Monitoring</a></li>
+          <li class="nested-sidebar-item"><a href="../deploying/ssl.html">Enabling SSL/TLS</a></li>
+          <li class="nested-sidebar-item"><a href="../deploying/docker.html">Docker</a></li>
+          <li class="nested-sidebar-item"><a href="../deploying/kubernetes.html">Kubernetes</a></li>
+          <li class="nested-sidebar-item"><a href="../deploying/cloud-foundry.html">Cloud Foundry</a></li>
+        </ul>
+        <li class="sidebar-item collapsible">Configuring</li>
+        <ul class="nested-sidebar-list content">
+          <li class="nested-sidebar-item"><a href="../configuring/enabling-nio.html">Enabling SwiftNIO</a></li>
+        </ul>
+      </ul>
+    </aside>
+    <div id="burgerIcon" class="burger-icon" onclick="showSidebar()">
+      <div class="burger-line"></div>
+      <div class="burger-line"></div>
+      <div class="burger-line"></div>
+    </div>
+    <div class="docs-item-2 search-container">
+    </div>
+    <nav class="docs-item-3 docs-nav">
+      <button id="api-button" class="apiref-button" type="button" name="button" onclick="window.open('https://ibm-swift.github.io/Kitura/')">API Reference</button>
+      <a class="nav-item" target="_blank" href="http://slack.kitura.io/">Need help?</a>
+      <a class="nav-item" target="_blank" href="https://github.com/IBM-Swift/kitura.io/issues">Found an issue?</a>
+    </nav>
+    <div id="doc-container" class="docs-item-4 docs-window">
+      <main>
+        <h1 class="heading-1">Kitura-WebSocket Echo Server</h1>
+        <p class="block-text">The application we are going to create is a very simple Echo Server, this is a server that will receive a message from the client and then echo back the same message to the client, this is the simplest implementation of WebSockets.</p>
```

Comment:
This is meant to be a guide but currently reads like a sample application. We need to try and explain to users why they are using specific bits of the WebSocket code.